src/components/MainApp.js

```diff
@@ -7,42 +7,56 @@ import useData from "../useData";
 
 const ALIAS_MATERIAS = require("../data/materias.json");
 
+const allCodigos = new Set(
+  Object.keys(ALIAS_MATERIAS).map((c) => c.toLowerCase()),
+);
+
 const MainApp = () => {
   const { data, partialLoading } = useData();
   const repos = React.useMemo(() => {
     return data.map((r) => ({
       user: r.owner.login,
       repoName: r.name,
       description: r.description,
+      codigos: new Set(r.topics).intersection(allCodigos),
       repoData: r,
     }));
   }, [data]);
 
-  const materias = React.useMemo(() => {
-    const mapa = data.reduce((mapa, repo) => {
-      const codigosEnRepo = repo.topics
-        .map((t) => t.toUpperCase())
-        .filter((t) => ALIAS_MATERIAS.hasOwnProperty(t));
-      codigosEnRepo.forEach((codigoEnRepo) => {
-        const nombreMateria = ALIAS_MATERIAS[codigoEnRepo];
-        const valuesAntes = mapa.get(nombreMateria);
-        const codigosAntes = valuesAntes ? valuesAntes.codigos : [];
-        const reposAntes = valuesAntes ? valuesAntes.reponames : [];
-        mapa.set(nombreMateria, {
-          codigos: [...new Set([...codigosAntes, codigoEnRepo])],
-          reponames: [...new Set([...reposAntes, repo.full_name])],
+  // Es un Map<string, {codigos, reponames}>
+  // Esto permite que se muestren todos los codigos de una materia con un mismo nombre
+  const materiasPorNombre = React.useMemo(() => {
+    const materiasPorNombre = new Map();
+    for (const [key, value] of Object.entries(ALIAS_MATERIAS)) {
+      if (!materiasPorNombre.has(value)) {
+        materiasPorNombre.set(value, {
+          codigos: new Set(),
+          reponames: new Set(),
         });
-      });
+      }
+      materiasPorNombre.get(value).codigos.add(key);
+    }
+    return materiasPorNombre;
+  }, []);
 
-      return mapa;
-    }, new Map());
-    const materias = Array.from(mapa, ([nombreMateria, objeto]) => ({
-      nombre: nombreMateria,
-      ...objeto,
+  const materias = React.useMemo(() => {
+    // Agregamos los reponames al materiasPorNombre
+    repos.forEach((repo) => {
+      repo.codigos.forEach((codigoEnRepo) => {
+        const nombreMateria = ALIAS_MATERIAS[codigoEnRepo.toUpperCase()];
+        const materiaMap = materiasPorNombre.get(nombreMateria)
+        materiaMap.reponames = materiaMap.reponames.add(repo.repoData.full_name)
```

Comment:
```suggestion
        materiasPorNombre.get(nombreMateria).reponames.add(full_name);
```

y te ahorras una variable intermedia?

Reply:
Ahh si tenes razón 🤦‍♂️ 
Me pasaba que con .union() no se actualizaba y pasaron cosas 🤣